fastzipfile.py

```diff
@@ -40,8 +42,73 @@ def _read2(self, n):
 
 
 def monkeypatch():
+    # This monkey patching makes a lot of assumptions on the zipfile module that are not
+    # part of the public interface and therefore might change at any time. To avoid
+    # errors caused by that, do some sanity checks first. Changes can be checked in the
+    # CPython repository. E.g.:
+    #   git clone https://github.com/python/cpython.git
+    #   git diff -w --color-moved origin/3.10:Lib/zipfile.py origin/3.11:Lib/zipfile.py
+    # Note that Lib/zipfile.py was split up and moved into lib/zipfile since Python 3.12.0a3!
+    # Of course, we are mostly only interested in changes to zipfile.ZipExtFile and
+    # zipfile._ZipyDecrypter.
+    if not hasattr(zipfile, "ZipExtFile"):
+        print(
```

Comment:
It would be better to use `warnings` module so that user has ability to turn off and re route to other streams than `stdout`